internal/controller/resourcesetinputprovider_controller.go

```diff
@@ -0,0 +1,399 @@
+			retErr = kerrors.NewAggregate([]error{retErr, err})
+		}
+	}()
+
+	// Uninstall if the object is under deletion.
+	if !obj.ObjectMeta.DeletionTimestamp.IsZero() {
+		// Release the object to be garbage collected.
+		controllerutil.RemoveFinalizer(obj, fluxcdv1.Finalizer)
+
+		// Stop reconciliation as the object is being deleted.
+		return ctrl.Result{}, nil
+	}
+
+	// Add the finalizer if it does not exist.
+	if !controllerutil.ContainsFinalizer(obj, fluxcdv1.Finalizer) {
+		log.Info("Adding finalizer", "finalizer", fluxcdv1.Finalizer)
+		controllerutil.AddFinalizer(obj, fluxcdv1.Finalizer)
+		conditions.MarkUnknown(obj,
+			meta.ReadyCondition,
+			meta.ProgressingReason,
+			"%s", msgInProgress)
+		conditions.MarkReconciling(obj,
+			meta.ProgressingReason,
+			"%s", msgInProgress)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	// Pause reconciliation if the object has the reconcile annotation set to 'disabled'.
+	if obj.IsDisabled() {
+		log.Error(errors.New("can't reconcile instance"), fluxcdv1.ReconciliationDisabledMessage)
+		r.Event(obj, corev1.EventTypeWarning, fluxcdv1.ReconciliationDisabledReason, fluxcdv1.ReconciliationDisabledMessage)
+		return ctrl.Result{}, nil
+	}
+
+	// Reconcile the object.
+	return r.reconcile(ctx, obj, patcher)
+}
+
+func (r *ResourceSetInputProviderReconciler) reconcile(ctx context.Context,
+	obj *fluxcdv1.ResourceSetInputProvider,
+	patcher *patch.SerialPatcher) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	reconcileStart := time.Now()
+
+	// Mark the object as reconciling.
+	msg := "Reconciliation in progress"
+	conditions.MarkUnknown(obj,
+		meta.ReadyCondition,
+		meta.ProgressingReason,
+		"%s", msg)
+	conditions.MarkReconciling(obj,
+		meta.ProgressingReason,
+		"%s", msg)
+	if err := r.patch(ctx, obj, patcher); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to update status: %w", err)
+	}
+
+	// Get the basic auth credentials.
+	username, password, err := r.getBasicAuth(ctx, obj)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get credentials: %w", err)
+	}
+
+	// Get the CA certificate.
+	certPool, err := r.getCertPool(ctx, obj)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get certificates: %w", err)
+	}
+
+	// Create the provider based on the object type.
+	provider, err := r.newGitProvider(ctx, obj, certPool, username, password)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to create provider: %w", err)
+	}
+
+	// Get the provider options.
+	exportedInputs, err := r.callProvider(ctx, obj, provider)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to call provider: %w", err)
+	}
+
+	// Update the object status with the exported inputs.
+	data, err := yaml.Marshal(exportedInputs)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to marshal exported inputs: %w", err)
+	}
+	obj.Status.ExportedInputs = exportedInputs
+	obj.Status.LastExportedRevision = digest.FromBytes(data).String()
```

Comment:
I assume `yaml.Marshal()` is deterministic, i.e. it sorts map keys?

Reply:
Yeap it's alphabetical sort